BER-395: Sync origin with upstream v0.27.13

denasescu opened the pull request:
**Context**
> Why do we want/need it?
- To keep origin up to date.

**Relevant issues**
- Closes https://github.com/nedap/berichten-ios/issues/395

**Changes**
> What does it do?
> In summary, what changes are made to the code?
- Merged upstream (MatrixSDK v0.27.13)
- Matrix adds `accessToken` to loading media, logic that we added some time ago. In this case I replaced our changes with Matrix changes (basically removing `Modified by Nedap` comments) because it's doing the same thing.
- I added the tag `nedap/0.1.4` to this branch to be able to test it in berichten-ios, but I will remove it and add it to `ons_master` once the branch is approved and merged. 

**Known problems**
- What problems do still arise or are anticipated?
- Which part of the solution do you need help with?

**Extra attention**
- What do I need to pay extra attention to as a reviewer?

**Check-list**
- [ ] Acceptance criteria described in the issue are satisfied, or the bug(s) is/are fixed.
- [ ] Documentation present (relevant information is properly documented).
- [ ] Written tests for test suite (optional).
- [ ] CHANGELOG updated.

**Tested on devices**
- [ ] Minimum supported iOS version.
- [ ] Maximum supported iOS version.
- [ ] Lowest resolution supported.
- [ ] Highest resolution supported.
- [ ] Device with a notch.
- [ ] Device without a notch.

**Is this issue considered as Done?**
- [ ] It meets our [DoD](https://github.com/nedap/healthcare-mobile/wiki/DoD).

giomfo committed:
Add a new state event type: "m.room.retention"

# Conflicts:
#	MatrixSDK/Contrib/Swift/JSONModels/MXEvent.swift
#	MatrixSDK/JSONModels/MXEvent.h
#	MatrixSDK/JSONModels/MXEvent.m
#	MatrixSDK/Utils/MXTools.m

Velin92 committed:
Merge branch 'master' into develop

Velin92 committed:
Prepare for new sprint

stefanceriu committed:
version++

stefanceriu committed:
Merge pull request #1844 from matrix-org/release/0.27.6/release

Release 0.27.6

stefanceriu committed:
finish version++

stefanceriu committed:
Merge branch 'release/0.27.6/master'

stefanceriu committed:
Merge branch 'master' into develop

stefanceriu committed:
Prepare for new sprint

BillCarsonFr committed:
Add own device creation ts in crypto API

BillCarsonFr committed:
deviceCreationTs should be readonly

BillCarsonFr committed:
Merge pull request #1848 from matrix-org/valere/expose_own_device_creation_ts

Add own device creation ts in crypto API

pixlwave committed:
version++

pixlwave committed:
Merge pull request #1850 from matrix-org/release/0.27.7/release

Release 0.27.7

pixlwave committed:
Merge branch 'release/0.27.7/master'

pixlwave committed:
finish version++

pixlwave committed:
Merge branch 'master' into develop

pixlwave committed:
Prepare for new sprint

NicolasBuquet committed:
Sorting room list alphabetically case-insensitive. (#1851)

* Sort Room list alphabetically but caseInsensitive

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>

* Sort Room list alphabetically but caseInsensitive

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>

* Sort Room list alphabetically but caseInsensitive

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>

---------

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>
Co-authored-by: Nicolas Buquet <nbuquet@buquet-net.com>

BillCarsonFr committed:
Bump rust crypto sdk version 0.4.1

BillCarsonFr committed:
update minimum ios platform 13.0

BillCarsonFr committed:
Update min deployment target to 13.0

BillCarsonFr committed:
Update osx min platform

BillCarsonFr committed:
update podfile lock

BillCarsonFr committed:
Fix test compilation

BillCarsonFr committed:
Add changelog

BillCarsonFr committed:
Merge pull request #1853 from matrix-org/valere/bump_crypto_sdk_version_0.4.1

Bump rust crypto sdk version 0.4.1

BillCarsonFr committed:
Invalidate crypto store cache when entering foreground

BillCarsonFr committed:
Bump crypto sdk to 0.4.2

BillCarsonFr committed:
Fix warnings

stefanceriu committed:
Bump the Crypto version in the podspec as well

stefanceriu committed:
Merge pull request #1854 from matrix-org/valere/NSE_partial_fix

Invalidate crypto store cache when entering foreground

stefanceriu committed:
version++

stefanceriu committed:
Merge pull request #1856 from matrix-org/release/0.27.8/release

Release 0.27.8

stefanceriu committed:
Merge branch 'release/0.27.8/master'

stefanceriu committed:
finish version++

stefanceriu committed:
Merge branch 'master' into develop

stefanceriu committed:
Prepare for new sprint

BillCarsonFr committed:
Fix | Share room keys with dehydrated devices with rust stack

stefanceriu committed:
Add RestClient method for reporting rooms through MSC4151

stefanceriu committed:
Merge pull request #1859 from matrix-org/stefan/reportRooms

Add RestClient method for reporting rooms through MSC4151

stefanceriu committed:
version++

stefanceriu committed:
Merge pull request #1860 from matrix-org/release/0.27.9/release

Release 0.27.9

stefanceriu committed:
Merge branch 'release/0.27.9/master'

stefanceriu committed:
finish version++

stefanceriu committed:
Merge branch 'master' into develop

stefanceriu committed:
Prepare for new sprint

Velin92 committed:
added some functions to remove messages

before a certain timestamp

Velin92 committed:
remove all messages sent

Velin92 committed:
fixing a compilation error

Velin92 committed:
Merge pull request #1861 from matrix-org/mauroromito/add_m.room.retention

Add a new state event type: "m.room.retention"

pixlwave committed:
version++

pixlwave committed:
Merge pull request #1862 from matrix-org/release/0.27.10/release

Release 0.27.10

pixlwave committed:
finish version++

pixlwave committed:
Merge branch 'release/0.27.10/master'

pixlwave committed:
Merge branch 'master' into develop

pixlwave committed:
Prepare for new sprint

pixlwave committed:
Update README. (#1863)

Velin92 committed:
fix store issue

Velin92 committed:
Merge pull request #1864 from matrix-org/mauroromito/fix_store_issue

Velin92 committed:
version++

Velin92 committed:
Merge pull request #1865 from matrix-org/release/0.27.11/release

Velin92 committed:
finish version++

Velin92 committed:
Merge branch 'release/0.27.11/master'

Velin92 committed:
Merge branch 'master' into develop

Velin92 committed:
Prepare for new sprint

NicolasBuquet committed:
Fix CallKit audio session late init

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>

Velin92 committed:
Merge pull request #1866 from NicolasBuquet/fix-callkit-audio-session-late-init

BillCarsonFr committed:
Merge pull request #1858 from matrix-org/valere/fix_sending_to_dehydrated_device

Fix | Share room keys with dehydrated devices with rust stack

NicolasBuquet committed:
Expose MXRoomPowerLevels Swift wrappers to Element (#1869)

Signed-off-by: Nicolas Buquet <nbuquet@buquet-net.com>
Co-authored-by: Nicolas Buquet <nbuquet@buquet-net.com>

Velin92 committed:
implemented for the preview url API

Velin92 committed:
removing return used for testing

Velin92 committed:
using rest client instead of the home server directly

Velin92 committed:
added the access token to the media loader request

Velin92 committed:
implementation based on the server version

Velin92 committed:
access token fix

and using the stable endpoint prefix

Velin92 committed:
documentation

Velin92 committed:
code improvement

Velin92 committed:
Merge pull request #1871 from matrix-org/mauroromito/content_auth_APIs

Velin92 committed:
version++

Velin92 committed:
Merge pull request #1873 from matrix-org/release/0.27.12/release

Velin92 committed:
finish version++

Velin92 committed:
Merge branch 'release/0.27.12/master'

Velin92 committed:
Merge branch 'master' into develop

Velin92 committed:
Prepare for new sprint

richvdh committed:
Add UTC timestamps to console logs

richvdh committed:
Create 7472.change

Commit:
Bring in line with foundation DCO (#1875)

* bring in line with foundation DCO

* add changelog entry

---------

Co-authored-by: Josh Simmons <git@josh.tel>

stefanceriu committed:
Update format to include the time, Z cannot be applied to NSLog timestamps and doesn't do anything by itself

stefanceriu committed:
Merge pull request #1874 from matrix-org/richvdh-patch-1

Add UTC timestamps to console logs

pixlwave committed:
Use the codecov token. (#1877)

* Use the codecov token.

* Use codecov v4?

pixlwave committed:
version++

pixlwave committed:
Merge pull request #1878 from matrix-org/release/0.27.13/release

Release 0.27.13

pixlwave committed:
finish version++

denasescu committed:
Merge remote-tracking branch 'upstream/master' into feature/395_sync_origin_with_upstream_v0_27_13

# Conflicts:
#	CHANGES.md
#	MatrixSDK.podspec
#	MatrixSDK/MXSession.m
#	MatrixSDK/MatrixSDKVersion.m
#	MatrixSDK/Utils/Media/MXMediaLoader.h
#	MatrixSDK/Utils/Media/MXMediaLoader.m
#	MatrixSDK/Utils/Media/MXMediaManager.h
#	MatrixSDK/Utils/Media/MXMediaManager.m
#	Podfile.lock

denasescu committed:
BER-395: update changelog

martias commented on MatrixSDK/Utils/Media/MXMediaLoader.m:
```diff
@@ -40,31 +40,22 @@
 NSString *const kMXMediaUploadIdPrefix = @"upload-";
 
-// Modified by Nedap. Store accessToken to set the authorization token on media requests (BER-229)
+
```
Should this comment be removed?

denasescu replied:
Yes, I added this comment for `accessToken` property and I removed it now because the same property (with the exact name) has been added by matrix 😄.  So, I kept matrix changes in this case.

martias commented on MatrixSDK/Utils/Media/MXMediaLoader.m:
```diff
@@ -40,31 +40,22 @@
-}
-
-// Modified by Nedap. Init with access token to set the authorization token on media requests (BER-229)
-- (id)initWithAccessToken:(NSString *)accessToken
```
please check this one as well

denasescu replied:
Yes, I removed it because matrix also added `(id)initWithAccessToken:(NSString *) accessToken` so I kept their changes 😄

martias commented on MatrixSDK/Utils/Media/MXMediaLoader.m:
```diff
@@ -165,11 +156,7 @@ - (void)downloadMediaFromURL:(NSString *)url
         [request setValue:value forHTTPHeaderField:key];
     }];
     
-    // Modified by Nedap. Set the authorization token on requests (BER-229)
```
Check this modification

denasescu replied:
Yes it's correct, because Matrix now adds `[request setValue: [NSString stringWithFormat:@"Bearer %@", _accessToken] forHTTPHeaderField: @"Authorization"];` (the same thing I was doing here), so I kept Matrix changes.

martias reviewed (commented):
It seemed to work in the app, but please double check that there is some Nedap code removed

denasescu commented:
> It seemed to work in the app, but please double check that there is some Nedap code removed

@martias  Thanks for the review! I was also surprised to see matrix now adding `accessToken` to the request, feature we implemented some time ago. So, I've removed our code and kept the matrix implementation since it is doing the same thing 😄

martias reviewed (approved):
Thanks for the reply to my comments!, lets go green @denasescu
